[MSE][GStreamer] Do more pipeline dumps in WebKitMediaSrc

ntrrgc opened the pull request:
#### b4ac73768e74d52bf877a1c466eeee4408f291c2
<pre>
[MSE][GStreamer] Do more pipeline dumps in WebKitMediaSrc
<a href="https://bugs.webkit.org/show_bug.cgi?id=270980">https://bugs.webkit.org/show_bug.cgi?id=270980</a>

Reviewed by Philippe Normand.

This patch adds more points where, if requested,
WebKitMediaSourceGStreamer will dump the playback pipeline.

It also refactors pipeline dumping into a separate function for improved
clarity and code reuse.

Pipeline dumps that were produced before by WebKitMediaSourceGStreamer:

- playback-pipeline-before-playback-{trackId}
- playback-pipeline-pushing-buffer-failed-{trackId}

Pipeline dumps that are produced after the patch:

- playback-pipeline-{trackId}-first-frame-before
- playback-pipeline-{trackId}-first-frame-after
- playback-pipeline-{trackId}-pushing-buffer-failed
- playback-pipeline-{trackId}-flush-start-before
- playback-pipeline-{trackId}-flush-start-after
- playback-pipeline-{trackId}-flush-stop-before
- playback-pipeline-{trackId}-flush-stop-after

Track ID has also been moved left in the name to bring attention to it.

* Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp:
(dumpPipeline):
(webKitMediaSrcLoop):
(webKitMediaSrcStreamFlush):

Canonical link: <a href="https://commits.webkit.org/276341@main">https://commits.webkit.org/276341@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2e9670e62832e45e35c8a3bd77fa52e6e99d663d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43927 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22981 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46357 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46702 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40001 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46231 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27068 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20379 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36330 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44502 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20170 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37834 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17344 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/17648 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38928 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1983 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40240 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39214 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48144 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18931 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15506 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43070 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20320 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41781 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9870 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20526 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19945 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp:
```diff
@@ -210,6 +210,22 @@ static GRefPtr<GstElement> findPipeline(GRefPtr<GstElement> element)
+{
+#ifdef GST_DISABLE_GST_DEBUG
+    (void) fileNamePattern;
+    (void) stream;
```
`[[maybe_unused]]` :)

philn commented on Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp:
```diff
@@ -210,6 +210,22 @@ static GRefPtr<GstElement> findPipeline(GRefPtr<GstElement> element)
+    (void) fileNamePattern;
+    (void) stream;
+#else
+    GUniquePtr<char> fileName {
```
Nit: might be nicer to use `makeString()`

ntrrgc committed:
[MSE][GStreamer] Do more pipeline dumps in WebKitMediaSrc
https://bugs.webkit.org/show_bug.cgi?id=270980

Reviewed by Philippe Normand.

This patch adds more points where, if requested,
WebKitMediaSourceGStreamer will dump the playback pipeline.

It also refactors pipeline dumping into a separate function for improved
clarity and code reuse.

Pipeline dumps that were produced before by WebKitMediaSourceGStreamer:

- playback-pipeline-before-playback-{trackId}
- playback-pipeline-pushing-buffer-failed-{trackId}

Pipeline dumps that are produced after the patch:

- playback-pipeline-{trackId}-first-frame-before
- playback-pipeline-{trackId}-first-frame-after
- playback-pipeline-{trackId}-pushing-buffer-failed
- playback-pipeline-{trackId}-flush-start-before
- playback-pipeline-{trackId}-flush-start-after
- playback-pipeline-{trackId}-flush-stop-before
- playback-pipeline-{trackId}-flush-stop-after

Track ID has also been moved left in the name to bring attention to it.

* Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp:
(dumpPipeline):
(webKitMediaSrcLoop):
(webKitMediaSrcStreamFlush):

Canonical link: https://commits.webkit.org/276341@main
